src/Platform/Microsoft.Testing.Platform/Configurations/JsonConfigurationProvider.cs

```diff
@@ -29,13 +35,26 @@ private async Task LogInformationAsync(string message)
 
         public async Task LoadAsync()
         {
-            string configFileName = $"{Path.Combine(
-                Path.GetDirectoryName(_testApplicationModuleInfo.GetCurrentTestApplicationFullPath())!,
-                Path.GetFileNameWithoutExtension(_testApplicationModuleInfo.GetCurrentTestApplicationFullPath()))}{PlatformConfigurationConstants.PlatformConfigSuffixFileName}";
-            if (!_fileSystem.Exists(configFileName))
+            string configFileName;
+            if (_commandLineParseResult.TryGetOptionArgumentList(PlatformCommandLineProvider.ConfigOptionKey, out string[]? configOptions))
             {
-                await LogInformationAsync($"Config file '{configFileName}' not found.");
-                return;
+                configFileName = Path.GetFullPath(configOptions[0]);
```

Comment:
wonder if we should avoid this api that could throw https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getfullpath?view=net-8.0 and simply test for existance

Reply:
Exists rule looks good to me

`The path parameter is permitted to specify relative or absolute path information. Relative path information is interpreted as relative to the current working directory. To obtain the current working directory, see GetCurrentDirectory.`

Reply:
My intent here was to provide an exception with the full path when it's not found.
So that the user can clearly see a relative path in `--config-file` is relative to "what".

Are you okay to evaluate GetFullPath only when we are about to throw an exception, and possible catch any exception during GetFullPath and do nothing?

Somthing like this?

```csharp
if (_commandLineParseResult.TryGetOptionArgumentList(PlatformCommandLineProvider.ConfigOptionKey, out string[]? configOptions))
{
    configFileName = configOptions[0];
    if (!_fileSystem.Exists(configFileName))
    {
        try
        {
            // Get the full path for better error messages.
            // As this is only for the purpose of throwing an exception, ignore any exceptions during the GetFullPath call.
            configFileName = Path.GetFullPath(configFileName);
        }
        catch
        {
        }

        // TODO: Localize
        throw new FileNotFoundException($"The configuration file '{configFileName}' specified with '--config-file' could not be found.", configFileName);
    }
}
```

Reply:
I think it sounds reasonable, let's get confirmation from @MarcoRossignoli.

Reply:
Ok fine to me, anyway I don't expect we will reach this point if not for some bug, I'd like to have this check on the verify option value, where we define the option.

Reply:
I added it there as well.